### PR TITLE
Hooks: update H5py

### LINKED ISSUE #2540: hook h5py issue
With the current develop branch (https://github.com/pyinstaller/pyinstaller/commit/483dfdebad65679ded59476514c9a5528c4ba3b6), on Win10 and Python 3.6, the hidden imports listed in the hook are not actually bundled.

I attempted to manually copy and rename the 4 pyds (https://github.com/pyinstaller/pyinstaller/blob/a70b20e4de6a6817987d28ca9f3201c8105fd858/PyInstaller/hooks/hook-h5py.py), and it works.

### PR DESCRIPTION
**Pull request by xoviat**
Closes #2540.

### CODE REVIEW

**Comment by xoviat** [comment]
This is less than optimal because the pyd files are duplicated. However, the names appear to be hard coded in the pyd files (probably by the linker).

**Comment by xoviat** [comment]
Go ahead and wait on this because I will try to fix the folder structure.

File: PyInstaller/loader/pyimod03_importers.py [hunk 1]
```diff
@@ -642,9 +656,13 @@ class CExtensionImporter(object):
         full.module.name.cpython-33m.so
         full.module.name.abi3.so
     """
+
     def __init__(self):
         # Cache directory content for faster module lookup without
         # file system access.
+
+        # TODO: need recursive read capability
+
```
Comment (xoviat) [inline PyInstaller/loader/pyimod03_importers.py]: We cannot use os.path.walk here. I am not sure how to build up recursive capability.

**Comment by htgoebel** [comment]
@xoviat I merged #2686, also regarding h5py, just a few days ago. Sorry, I missed your pull-request.

Questions:
- #2686 adds far less modules. Are all these modules your hook is adding required?
- The original hook (and your version, too) adds '_proxy', 'utils', 'defs', which I assume are no valid modules. Shouldn't these be removed?

Please rebase your pull-request, so I can merge it, if you think it's ready (again).

**Comment by xoviat** [comment]
The other pull request should resolve these errors.

**Pull request closed without merging**